src/components/Tools/ProgressTracker.vue

```diff
@@ -0,0 +1,214 @@
+<template>
+  <div class="progress">
+    <div class="progress-position">
+      <div class="progress-bar-container">
+        <div class="progress-bar-overflow">
+          <div class="progress-bar" :style="progressBarStyle"></div>
+        </div>
+        <img class="progress-bar-caption" :src="getImage" />
+      </div>
+      <div class="progress-status">
+        <span class="progress-status-text">{{ getProgressString }}</span>
+      </div>
+    </div>
+    <div class="progress-text">
+      <span class="progress-text-style"> You've completed: </span>
+      <div v-for="(req, index) in requirementProgressBundles" :key="index" class="progress-row">
+        <span class="progress-numfulfilled progress-text-style"
+          >{{ Math.floor(req.safeProgress) }} / {{ req.totalRequired }}</span
+        >
+        <span class="progress-reqname progress-text-style"> {{ req.groupName }} Requirements </span>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import store from '@/store';
+import confetti from '@/assets/images/progress_tracker/confetti.svg';
+import fire from '@/assets/images/progress_tracker/fire.svg';
+import flex from '@/assets/images/progress_tracker/flex.svg';
+import hands from '@/assets/images/progress_tracker/hands_raised.svg';
+import star from '@/assets/images/progress_tracker/star.svg';
+import {
+  groupedRequirementDangerouslyFulfilled,
+  groupedRequirementTotalDangerousRequirementProgress,
+  groupedRequirementTotalRequired,
+  groupedRequirementTotalSafeRequirementProgress,
+} from '@/requirements/requirement-frontend-computation';
+
+export default defineComponent({
+  computed: {
+    requirementProgressBundles(): GroupedRequirementFulfillmentReportWithProgress[] {
```

Comment:
I feel like this shouldn't be in the Vue component

Reply:
I initially had all the different properties bundled into one function, but I got worried that that would cause performance issues with the req sidebar because only parts of it are used at a time.